### PR TITLE
User sign in

### PR DESCRIPTION
**Pull request by MIsabelZM**

### CODE REVIEW

**Comment by mixergtz** [comment]
Super :) @MIsabelZM  esperemos que @orendon  le de una miradita y hacemos el merge

File: app/assets/javascripts/task_lists.js.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (orendon) [inline app/assets/javascripts/task_lists.js.coffee]: si no se van a usar estos archivos ni los css ni los test se pueden borrar

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    @task_list = TaskList.find(params[:task_list_id])
+    @tasks = @task_list.tasks
+  end
+
+  def show
+  end
+
+  def new
+    @task_list = TaskList.find(params[:task_list_id])
+    @task = @task_list.tasks.new
```
Comment (orendon) [inline app/controllers/tasks_controller.rb]: how is this better than `Task.new`? 
did you mean `@task_list.tasks.build`?

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+class TasksController < ApplicationController
+  before_action :set_task, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @task_list = TaskList.find(params[:task_list_id])
+    @tasks = @task_list.tasks
```
Comment (orendon) [inline app/controllers/tasks_controller.rb]: we can get ride of `@tasks` and use `@task_list.tasks.each` on index view

also we can use `includes(:tasks)` so we preload this association avoiding n+1 query's

File: app/views/task_lists/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<%= form_for(@task_list) do |f| %>
+  <% if @task_list.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@task_list.errors.count, "error") %> prohibited this task_list from being saved:</h2>
```
Comment (orendon) [inline app/views/task_lists/_form.html.erb]: this seems to be duplicated across several forms.

what about moving this to `shared/errors` and refactor it so the given model gets passed as a partial local variable?

File: config/initializers/devise.rb [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # config.secret_key = 'e9265a62023e40930791c6d0262a4fabb3132215eac8b10f4317381bc784c1869aea851bdd33857934fef915928a7dbc1f39778881452ed0c82b81564c357d5a'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (orendon) [inline config/initializers/devise.rb]: :trollface:

**Comment by orendon** [comment]
let's setup travis so we can see this fancy test <s>failing</s> passing :trollface:

File: app/models/task.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+
+  enum status: [:completed, :pending]
+
+  protected
+
+  def set_default_status
+    self.status = 1
+  end
+
```
Comment (orendon) [inline app/models/task.rb]: no validations?

File: app/models/task_list.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  before_save :set_default_status
+
+  has_many :tasks
+  belongs_to :user
+  enum status: [:completed, :pending]
+
+  protected
+
+  def set_default_status
+    self.status = 1
```
Comment (orendon) [inline app/models/task_list.rb]: you can use `self.status = :pending` as well

File: app/models/task_list.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class TaskList < ActiveRecord::Base
+  before_save :set_default_status
```
Comment (orendon) [inline app/models/task_list.rb]: prolly an after_initialize callback would be better, do we have an unit test for this?

File: app/views/devise/passwords/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<div class="container">
+  <h2>Change your password</h2>
+
+  <%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= devise_error_messages! %>
+    <%= f.hidden_field :reset_password_token %>
+
+    <div><%= f.label :password, "New password" %><br />
```
Comment (orendon) [inline app/views/devise/passwords/edit.html.erb]: I believe we're missing some bootstrap classes across devise form fields....

File: app/views/task_lists/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        <h3>Current Lists</h3>
+      </div>
+      <div class="col-md-1 text-center">
+        <%= link_to raw('<i class="fa fa-plus-circle fa-4x"></i>'), new_task_list_path %>
+      </div>
+    </div>
+
+  <div class="row">
+    <div class="col-md-10 col-md-offset-1">
+      <%= render @task_lists %>
```
Comment (orendon) [inline app/views/task_lists/index.html.erb]: :+1:

**Comment by orendon** [comment]
looking good! just left some minor suggestions and refactoring, well done

**Comment by mixergtz** [comment]
Listas las correcciones, o que opina @orendon ?

**Comment by orendon** [comment]
:+1: :shipit:

**Pull request closed without merging**